### DISCUSSION TITLE
IP routing message on fresh install .140

### DISCUSSION THREAD

**Discussion by isaacgolding** [Security-Onion-Solutions/securityonion#8331] [Unsupported Versions]
So I'm installing .140 on a fresh VM using the exact same setup as my working .130 setup.  

After setting up the static ip address in the install screen I get the following message:
The IP being routed by linux is not the ip address assigned to the management interface ens18.  

![Capture](https://user-images.githubusercontent.com/104205381/179811546-1d7cac84-08bf-414b-953a-34d06606eb8b.JPG)

I drop down to the terminal interface and run ip a and there is the ip I just assigned.

2: ens18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether be:c7:1f:a7:e7:a9 brd ff:ff:ff:ff:ff:ff
    inet 10.23.83.240/24 brd 10.23.83.255 scope global noprefixroute ens18
       valid_lft forever preferred_lft forever
    inet6 fe80::400d:fe3e:fe22:1a4c/64 scope link noprefixroute
       valid_lft forever preferred_lft forever

If I rerun setup and attempt to setup using DHCP I get the same end result.

Any suggestions?

**Comment by dougburks** [comment]
Here's the code:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/setup/so-functions#L963-L984

Take a look at your `/root/sosetup.log` for the log entry at https://github.com/Security-Onion-Solutions/securityonion/blob/master/setup/so-functions#L966 to see if that provides any additional clues.

**Reply by isaacgolding** [reply]
So I won't claim to be a decent programmer :-(   but I can read the code enough to see that there isn't a lot going on there.     
I dug into /root/sosetup.log and all I see are more references to the same error message: error "[ERROR] Main gateway ($MAINIP) does not match ip address of management NIC ($MNIC_IP)."

1) I'm using ens18 as my management interface.   It's verified via the mac address ending in :a9

2) referencing the so-function line  # 884 I can see you are looking at the ip link command to get the list of interfaces.	

> readarray -t nic_list <<< "$(ip link| awk -F: '$0 !~ "lo|vir|veth|br|docker|wl|^[^0-9]"{print $2}' | grep -vwe "bond0"  | sed 's/ //g' | sed -r 's/(.*)(\.[0-9]+)@\1/\1\2/g')"

So I took the command  "ip link" and ran it at the terminal as root.  I get the following:

> [root@apollo isaac]# ip link
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
>     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
> 2: ens19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
>     link/ether b2:78:f4:0d:fc:c9 brd ff:ff:ff:ff:ff:ff
> 3: ens20: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
>     link/ether 4a:c6:e7:12:87:99 brd ff:ff:ff:ff:ff:ff
> 4: ens21: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
>     link/ether 1e:5f:23:22:65:a7 brd ff:ff:ff:ff:ff:ff
> 5: ens22: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
>     link/ether b6:f9:24:76:bf:97 brd ff:ff:ff:ff:ff:ff
> **7: ens18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
>     link/ether be:c7:1f:a7:e7:a9 brd ff:ff:ff:ff:ff:ff**

Follow that with "ip a" and I get the following:

> [root@apollo isaac]# ip a
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
>     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
>     inet 127.0.0.1/8 scope host lo
>        valid_lft forever preferred_lft forever
> 2: ens19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
>     link/ether b2:78:f4:0d:fc:c9 brd ff:ff:ff:ff:ff:ff
>     inet 10.23.83.123/24 brd 10.23.83.255 scope global noprefixroute dynamic ens19
>        valid_lft 26391sec preferred_lft 26391sec
> 3: ens20: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
>     link/ether 4a:c6:e7:12:87:99 brd ff:ff:ff:ff:ff:ff
>     inet6 fe80::2947:1321:d7dc:d500/64 scope link noprefixroute
>        valid_lft forever preferred_lft forever
> 4: ens21: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
>     link/ether 1e:5f:23:22:65:a7 brd ff:ff:ff:ff:ff:ff
>     inet6 fe80::d851:af78:f7e5:56cf/64 scope link noprefixroute
>        valid_lft forever preferred_lft forever
> 5: ens22: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
>     link/ether b6:f9:24:76:bf:97 brd ff:ff:ff:ff:ff:ff
>     inet6 fe80::a51f:d97:df64:145f/64 scope link noprefixroute
>        valid_lft forever preferred_lft forever
> **7: ens18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
>     link/ether be:c7:1f:a7:e7:a9 brd ff:ff:ff:ff:ff:ff
>     inet 10.23.83.240/24 brd 10.23.83.255 scope global noprefixroute ens18**
>        valid_lft forever preferred_lft forever
>     inet6 fe80::400d:fe3e:fe22:1a4c/64 scope link noprefixroute
>        valid_lft forever preferred_lft forever


I'm thinking the problem might lay right there.   But I definitely don't have all of the pieces.   Tomorrow I'll try to dig around in the OS logs and dmesg to see if I can find any other flags/issues/errors that are there and possibly related.

**Reply by dougburks** [reply]
Try removing the IP address from ens19:
> 2: ens19: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
link/ether b2:78:f4:0d:fc:c9 brd ff:ff:ff:ff:ff:ff
inet 10.23.83.123/24
